src/main/java/net/minestom/server/utils/chunk/ChunkUtils.java

```diff
@@ -81,4 +83,43 @@ public static Chunk retrieve(Instance instance, Chunk originChunk, double x, dou
     public static Chunk retrieve(Instance instance, Chunk originChunk, Point position) {
         return retrieve(instance, originChunk, position.x(), position.z());
     }
+
+    /**
+     * Computes the effective view distance based on the client's view distance and the target instance's view distance.
+     * The effective view distance is the minimum of the client's view distance and the instance's view distance, plus one.
+     *
+     * <p>Examples:
+     * <ul>
+     *   <li>Client VD = 10, Instance VD = 16 → min(10, 16) + 1 = 11</li>
+     *   <li>Client VD = 16, Instance VD = 8 → min(16, 8) + 1 = 9</li>
+     *   <li>Client VD = 12, Instance = null → min(12, ServerFlag.CHUNK_VIEW_DISTANCE) + 1</li>
+     * </ul>
+     *
+     * @param clientViewDistance the client's view distance setting
+     * @param targetInstance the target instance, or null to use the default server view distance
+     * @return the effective view distance
+     */
+    public static int computeEffectiveViewDistance(byte clientViewDistance, @Nullable Instance targetInstance) {
+        int maxViewDistance = targetInstance != null ? targetInstance.viewDistance() : ServerFlag.CHUNK_VIEW_DISTANCE;
+        return Math.min(clientViewDistance, maxViewDistance) + 1;
+    }
+
+    /**
+     * Computes the server view distance for the given instance, clamped between 2 and 32.
+     *
+     * <p>Examples:
+     * <ul>
+     *   <li>Instance VD = 10 → 10 (within bounds)</li>
+     *   <li>Instance VD = 1 → 2 (clamped to minimum)</li>
+     *   <li>Instance VD = 50 → 32 (clamped to maximum)</li>
+     *   <li>Instance = null → clamp(ServerFlag.CHUNK_VIEW_DISTANCE, 2, 32)</li>
+     * </ul>
+     *
+     * @param targetInstance the target instance, or null to use the default server view distance
+     * @return the server view distance, clamped between 2 and 32
+     */
+    public static int computeServerViewDistance(@Nullable Instance targetInstance) {
+        int viewDistance = targetInstance != null ? targetInstance.viewDistance() : ServerFlag.CHUNK_VIEW_DISTANCE;
+        return MathUtils.clamp(viewDistance, 2, 32);
```

Comment:
as kerman said, no need for an upper bound.
```suggestion
        return Math.max(2, viewDistance);
```